ui/redux/selectors/content.js

```diff
@@ -77,11 +80,39 @@ export const makeSelectNextUnplayedRecommended = (uri: string) =>
   createSelector(
     makeSelectRecommendedContentForUri(uri),
     selectHistory,
-    (possibleNext, history) => {
-      if (possibleNext) {
-        for (let i = 0; i < possibleNext.length; i++) {
-          if (!history.find(item => item.uri === possibleNext[i])) {
-            return possibleNext[i];
+    selectClaimsByUri,
+    selectAllCostInfoByUri,
+    selectBlockedChannels,
+    (
+      recommendedForUri: Array<string>,
+      history: Array<{ uri: string }>,
+      claimsByUri: { [string]: ?Claim },
+      costInfoByUri: { [string]: { cost: 0 | string } },
+      blockedChannels: Array<string>
+    ) => {
+      if (recommendedForUri) {
```

Comment:
Re-writing `for` loops into functional equivalents is the closest I get to zen meditation. Thank you for today's edition.

More importantly, you were using `find` when you probably want to use `some`.

```
return recommendedForUri.find(recommendedUri => {
  const { isChannel } = parseURI(recommendedUri);
  if (isChannel) {
    return false;
  }

  const costInfo = costInfoByUri[recommendedUri];
  if (!costInfo || costInfo.cost !== 0) {
    return false;
  }

  const claim = claimsByUri[recommendedUri];
  const channel = claim && claim.signing_channel;
  if (channel && blockedChannels.some(blockedUri => blockedUri === channel.permanent_url)) {
    return false;
  }

  return !history.some(item => item.uri === recommendedUri);
});
```